src/screens/SubmitRatingScreen.tsx

```diff
@@ -11,18 +11,26 @@ export interface SubmitRatingScreenProps {
 }
 
 export const DEFAULT_INTERIM_RATING = 3;
+const MAX_COUNT = 5;
+const MIN_COUNT = 0;
+const INCREMENT_VAL = 0.5;
 
 const SubmitRatingScreen: React.FC<SubmitRatingScreenProps> = ({
 	placeName,
 	photo_reference,
 }: SubmitRatingScreenProps) => {
+	const [counter0, setCounter0] = useState<number>(DEFAULT_INTERIM_RATING);
+	const [counter1, setCounter1] = useState<number>(DEFAULT_INTERIM_RATING);
+	const [counter2, setCounter2] = useState<number>(DEFAULT_INTERIM_RATING);
+	const [counter3, setCounter3] = useState<number>(DEFAULT_INTERIM_RATING);
```

Comment:
i feel like you're trying to go for a combination of dynamically creating counters based on PLACE_ATTRIBUTES, but hitting the rules of hooks and then statically hardcoding these hooks and dynamically mapping the place attributes with `counterStates` and indexes and everything

I think we would hit a better balance there by making one state for all the counters, and that would allow you to generate the state object dynamically from the place attributes.

for example:
```
useState(PLACE_ATTRIBUTES.reduce((countersMap, nextAttr) => countersMap[nextAttr] = DEFAULT_INTERIM_RATING, {})
```

Reply:
that will set the state to an object with fields for each of the ratings from the imported constant